### PR TITLE
Add a simple Kubernetes registry

### PR DESCRIPTION
**Pull request by Pryz**
This PR adds a really simple Kubernetes registry which will expose all the running PODs in the cluster to `/pods`.

The registry uses a watcher so it should not put much pressure on the Kubernetes cluster.

### CODE REVIEW

File: cmd/pprof-server/main.go [hunk 1]
```diff
@@ -120,3 +144,24 @@ type hostAddr string
...
+	if inCluster {
+		events.Log("kubernetes: using pod service account with InClusterConfig.")
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		events.Log("kubernetes: kubeconfig file.")
+		config, err = clientcmd.BuildConfigFromFlags(master, kubeconfig)
+		if err != nil {
```
Comment (achille-roussel) [inline cmd/pprof-server/main.go]: You could avoid repeating this `if err != nil` if you were moving it after the if/else block.

File: handler.go [hunk 2]
```diff
@@ -49,8 +52,11 @@ func (h *Handler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 	case path == "/services/":
 		h.serveListServices(res, req)
 
+	case strings.HasPrefix(path, "/pods"):
+		h.serveListPods(res, req)
```
Comment (achille-roussel) [inline handler.go]: Is your intention to keep this model permanently? If not then leaving a comment explaining the rough plan you have in mind would help whomever will be taking care of finalizing this work later on.

File: cmd/pprof-server/main.go (hunk 1 above)
Comment (Pryz) [reply cmd/pprof-server/main.go]: done.

File: handler.go (hunk 2 above)
Comment (Pryz) [reply handler.go]: done.

**Pull request merged**